src/basset.cpp

```diff
@@ -29,6 +29,46 @@ using std::unordered_set;
 using std::vector;
 using std::literals::string_literals::operator""s;
 
+string json_escape(const string &str) {
+  static const char hex[] = "0123456789abcdef";
+  std::string s;
+
+  for (const auto c : str) {
+    if (c == '\\' || c == '"') {
+      s.push_back('\\');
+      s.push_back(c);
+      continue;
+    }
+    if (c < 0x20) {
```

Comment:
Does not seem to work for UTF-8 inputs, at least if `char` is signed. Non-ASCII characters are represented in UTF-8 as multi-byte sequences, with every byte having its highest bit set. Before `<`, `c` is implicitly casted to `int` and the result will be negative, meaning that `c < 0x20` evaluates to `true` for every byte of UTF-8 multi-byte sequence, which is not what we want.